src/hooks/use-debounce.ts

```diff
@@ -1,6 +1,6 @@
 import { useEffect, useState } from "react"
 
-export function useDebounce(value, delay) {
+export const useDebounce = <T extends unknown>(value: T, delay: number): T => {
```

Comment:
suggestion: Would it make sense to do the following instead? since they are pretty much equivalent (except some cases) I think it simplify the hook
```ts
export const useDebounce = <T>(value: T, delay: number): T => {
```